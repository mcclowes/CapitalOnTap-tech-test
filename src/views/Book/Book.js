import Nav from "../Nav";
import React from "react";
import Product from "../../components/Product";
import { Page, PageWrapper } from "./csx";
import { Text } from "../../components/UIKit";

const Book = props => {
  return (
    <>
      <Nav />

      <PageWrapper>
        <Page>
          <Product id={props.bookId} />
        </Page>
      </PageWrapper>
    </>
  );
};

export default Book;
