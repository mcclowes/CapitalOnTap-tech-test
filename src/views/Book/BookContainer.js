import React, { useState } from "react";
import Book from "./Book";
import { Loader } from "../../components/UIKit";

const BookContainer = props => {
  const { match } = props;

  return <Book bookId={match && match.params.book} />;
};

export default BookContainer;
