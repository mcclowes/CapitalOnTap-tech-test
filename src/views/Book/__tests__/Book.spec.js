import Book from "../Book";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = newProps => {
  const props = {
    ...defaultProps,
    ...newProps
  };

  return shallow(<Book {...props} />);
};

let wrapper;

it("renders component", () => {
  wrapper = renderComponent();

  expect(wrapper).toMatchSnapshot();
});
