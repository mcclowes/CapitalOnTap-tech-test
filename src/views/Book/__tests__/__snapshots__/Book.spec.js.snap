// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<Fragment>
  <Nav />
  <csx__PageWrapper>
    <csx__Page>
      <Connect(ProductContainer) />
    </csx__Page>
  </csx__PageWrapper>
</Fragment>
`;
